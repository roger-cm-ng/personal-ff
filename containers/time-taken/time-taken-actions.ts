export const TIME_ADDED = 'TIME_ADDED';

export const addTime = (duration: number) => ({
  type: TIME_ADDED,
  duration
});
