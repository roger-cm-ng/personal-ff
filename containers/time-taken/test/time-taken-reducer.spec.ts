import timeTakenReducer from '../time-taken-reducer';
import { TIME_ADDED } from '../time-taken-actions';

describe('time taken reducer', () => {
  const initialState = () => timeTakenReducer(0, {
    type: TIME_ADDED,
    duration: 0
  });

  const acquiredState = () => timeTakenReducer(initialState(), {
    type: TIME_ADDED,
    duration: 200
  });

  it('should return initial state', () => {
    expect(initialState()).toEqual(0);
  });

  it('should update the state with acquired data when type is VEHICLES_FETCHED', () => {
    expect(acquiredState()).toMatchSnapshot();
    expect(acquiredState()).toEqual(200);
  });
});