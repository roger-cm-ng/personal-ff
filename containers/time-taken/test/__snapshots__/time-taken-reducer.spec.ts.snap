// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`time taken reducer should update the state with acquired data when type is VEHICLES_FETCHED 1`] = `200`;
